src/librustc_mir/interpret/eval_context.rs

```diff
@@ -838,7 +855,13 @@ impl<'a, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M
 
         match layout.variants {
             layout::Variants::Single { index } => {
-                return Ok(index as u128);
+                if layout.for_variant(&self, index).abi == layout::Abi::Uninhabited {
```

Comment:
This is intended to check that the next operation is valid, right? That means it should check `layout.abi`, not the variant's `.abi`.

Reply:
Done.